apps/website/screens/common/CodeBlock.tsx

```diff
@@ -0,0 +1,69 @@
+import { ReactNode, useState } from "react";
+import { DxcButton, DxcContainer, DxcFlex, HalstackProvider } from "@dxc-technology/halstack-react";
+import styled from "styled-components";
+
+const PreformattedText = styled.pre`
+  position: relative;
+  box-sizing: border-box;
+  margin: 0;
+  border-radius: 0.25rem;
+  padding: 1rem 1rem 1rem 1.5rem;
+  background-color: #333;
+`;
+
+const Code = styled.code`
+  font-size: 0.875rem;
+  white-space: pre-wrap;
+  word-wrap: break-word;
+  color: #fff;
+`;
+
+const buttonTokens = {
+  button: {
+    tertiaryDefaultFontColor: "#fff",
+    tertiaryHoverDefaultBackgroundColor: "#4d4d4d",
+    tertiaryActiveDefaultBackgroundColor: "#666666",
+  },
+};
+
+export default function CodeBlock({ children }: { children: ReactNode; language?: string }) {
+  const [copyActionIsVisible, setCopyActionIsVisible] = useState(false);
+  const [copied, setCopied] = useState(false);
+
+  const onMouseEnter = () => {
+    setCopyActionIsVisible(true);
+  };
+  const onMouseLeave = () => {
+    setCopyActionIsVisible(false);
+    setCopied(false);
+  };
+  const copyCode = () => {
```

Comment:
This is cool. However I think we could add a few minor features:
- When hovering and pressing Ctrl + C, it would be interesting to trigger the `copyCode`.
- It would be much more visually appealing to have it display the tooltip while the "Copied" message is active.

Reply:
- About the first point, I don't think It makes much sense. I find it very unintuitive since you do not have any visual clue of what is currently selected.
- How can you do that?